tests/spec/puppeteer-spec.js

```diff
@@ -174,4 +189,32 @@ describe("Puppeteer basic tests for the Python Editor.", function() {
 
         expect(fileExists).toBeTruthy();
     });
+
+    it("Correctly handles an mpy file", async function(){
+        const page = await browser.newPage();
+        await page.goto("http://localhost:5000/editor.html");
+        await page.click("#command-files");
+        let fileInput = await page.$("#file-upload-input");
+        await fileInput.uploadFile("./spec/test-files/samplempyfile.mpy");
+        await page.waitFor(500);
```

Comment:
Have you tried running the test without this waitFor? Maybe it can be removed (same for the one in line 213).